airflow/providers/amazon/aws/transfers/snowflake_to_s3.py

```diff
@@ -0,0 +1,150 @@
+        the method 'Configuring AWS IAM User Credentials' for unloading data to s3. If this is passed,
+        s3_bucket can't be used and file_format is not necessary
+    :type stage: str
+    :param warehouse: reference to a specific snowflake warehouse to override the one in the conn
+    :type warehouse: str
+    :param database: reference to a specific snowflake database to override the one in the conn
+    :type warehouse: str
+    :param s3_bucket: reference to a specific S3 bucket where the data will be saved. For using it, you
+        should have done the one-time setup 'Configuring a Snowflake Storage Integration'
+    :type s3_bucket: str
+    :param s3_key: reference to a specific S3 key within a bucket or stage.
+    :type s3_key: str
+    :param file_format: can be either a previous file format created in Snowflake
+        or hardcoded one like ``type = csv field_delimiter = ',' skip_header = 1``
+    :type file_format: str
+    :param query_or_table: query or full table to unload. If table, it must include the schema like
+        `schema.table`
+    :type query_or_table: str
+    :param snowflake_conn_id: reference to a specific snowflake connection
+    :type snowflake_conn_id: str
+    :param unload_options: reference to a list of UNLOAD options (SINGLE, MAX_FILE_SIZE,
+        OVERWRITE etc). Each element of the list has to be a string
+    :type unload_options: list
+    :param role: name of role (will overwrite any role defined in
+        connection's extra JSON)
+    :type role: str
+    :param authenticator: authenticator for Snowflake.
+        'snowflake' (default) to use the internal Snowflake authenticator
+        'externalbrowser' to authenticate using your web browser and
+        Okta, ADFS or any other SAML 2.0-compliant identify provider
+        (IdP) that has been defined for your account
+        'https://<your_okta_account_name>.okta.com' to authenticate
+        through native Okta.
+    :type authenticator: str
+    :param session_parameters: You can set session-level parameters at
+        the time you connect to Snowflake
+    :type session_parameters: dict
+    """
+
+    template_fields = (
+        's3_key',
+        's3_bucket',
+        'table_or_query',
+        'snowflake_conn_id',
+    )
+    template_ext = ('.sql',)
+    ui_color = '#ffebb2'
+
+    @apply_defaults
+    def __init__(
+        self,
+        *,
+        stage: Optional[str] = None,
+        s3_bucket: Optional[str],
+        s3_key: str,
+        query_or_table: str,
+        file_format: Optional[str],
+        database: Optional[str] = None,
+        warehouse: Optional[str] = None,
+        snowflake_conn_id: str = "snowflake_default",
+        unload_options: Optional[list] = None,
+        autocommit: bool = True,
+        include_header: bool = False,
+        role: Optional[str] = None,
+        authenticator: Optional[str] = None,
```

Comment:
i think it might help readability (and be more future-proof) if we consolidate hook kwargs to a `hook_kwargs` param.  What do y'all think?

This would separate the configuration necessary for the functioning of this operator from the configuration that _should_ be handled by airflow connection but may be overridden.  E.g. i am thinking session_parameters and authenticator would go in hook kwargs dict.   autocommit would be another but curiously it's not a hook param. (that said i don't think autocommit would have any effect whatsoever on an unload operation so can probably be removed here)

Reply:
I don't have a strong opinion on that. 

I will remove autocommit though